runtime/major_gc.c

```diff
@@ -1160,7 +1168,11 @@ static intnat mark_stack_push_block(struct mark_stack* stk, value block)
   end = (block_wsz < 8 ? block_wsz : 8);
 
   for (i = offset; i < end; i++) {
-    value v = Field(block, i);
+    /* This load may race with a concurrent caml_modify. It does not
+       constitute a data race as this is a volatile load. However, TSan will
+       wrongly see a race here (see section 3.2 of comment in tsan.c). We
+       therefore make sure it is never TSan-instrumented. */
+    value v = volatile_load_uninstrumented(&Field(block, i));
```

Comment:
I'm not fond of the redundancy introduced by duplicating the comment. Could you move the comment to the helper function definition, instead of repeating it at the callsite? (It might also be an occasion to clarify it a bit: if I understand correctly, the point is that these loads may race with mutator access, and/but that we have decided that these races are benign. The "does not constitute a data race" part of the comment is a bit confusing to me, I think it suggests that this is not UB?)

Reply:
(gentle ping)

Reply:
I have tried to clarify in deae015.